# Semana 4 · Diagnóstico y limpieza inicial de datos (Python)

**Práctica obligatoria · 2 pomodoros (40 minutos)**

En este notebook vas a diagnosticar problemas de calidad, trabajar sobre una copia de los datos y documentar cada decisión. No solo deberás escribir el código, sino también interpretar el problema y justificar el tratamiento.

> Realizá esta versión o la versión en R. No es necesario completar ambas.

**Resultados esperados**

Al finalizar deberías poder:

- revisar estructura, tipos, valores faltantes y duplicados;
- distinguir errores verificables de valores atípicos potencialmente válidos;
- registrar problemas, decisiones y justificaciones en una bitácora.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

pd.set_option("display.max_columns", None)

# Práctica Resuelta

## Dataset

Trabajaremos con el dataset 'Titanic'. Este conjunto de datos registra información sobre los pasajeros del trágico viaje del RMS Titanic en 1912. Este dataset se usa para realizar el análisis de datos y estudiar las características comunes de los pasajeros que sobrevivieron al accidente. Pero tiene un problema, contiene una gran variedad de errores y problemas. Por lo tanto, hay que realizar la limpieza de datos.

Aquí se describen en detalle las variables del dataset:

| Variable      | Descripción                                                                                                        |
| :------------ | :----------------------------------------------------------------------------------------------------------------- |
| `PassengerId` | Identificador único del pasajero.                                                                                  |
| `Survived`    | Indica si el pasajero sobrevivió (1) o no (0).                                                                     |
| `Pclass`      | Clase del pasaje (1ª, 2ª o 3ª clase). Es un indicador del estatus socioeconómico.                                     |
| `Name`        | Nombre del pasajero.                                                                                               |
| `Sex`         | Género del pasajero (male/female).                                                                                 |
| `Age`         | Edad del pasajero en años. Los valores fraccionarios representan edades estimadas.                                   |
| `SibSp`       | Número de hermanos/cónyuges a bordo del Titanic.                                                                   |
| `Parch`       | Número de padres/hijos a bordo del Titanic.                                                                        |
| `Ticket`      | Número de ticket.                                                                                                  |
| `Fare`        | Tarifa pagada por el pasaje.                                                                                       |
| `Cabin`       | Número de cabina. Muchos valores faltantes.                                                                       |
| `Embarked`    | Puerto de embarque (C = Cherbourg, Q = Queenstown, S = Southampton).                                               |

Con la siguiente celda podrás obtener el dataset completo. Si llega a generar errores o problemas, podés descargar el dataset que está subido en el aula virtual, subirlo a la carpeta "Archivos" de colab y ejecutar el siguiente código:

```
df_titanic = pd.load_csv('\content\titanic.csv')
```


In [2]:
url_titanic_raw = 'https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv'
df_titanic = pd.read_csv(url_titanic_raw)

df_titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


## 2. Diagnóstico inicial

Antes de modificar, debemos observar la cantidad de registros que tienen, los tipos de variables (columnas) y un resumen estadístico sobre el comportamiento de cada una.

### Tamaño del dataset

In [3]:
print(df_titanic.shape)

(891, 12)


Posee 891 filas y 12 columnas

### Tipos de variables

In [4]:
df_titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


Observamos que el dataset tiene 891 filas en total. Contiene 12 variables: 2 de tipo float (cuantitativas continuas), 5 de tipo int (cuantitativas, posiblemente discretas) y 5 de tipo object (cualitativas nomilase y ordinales).

También observamos la cantidad de valores no nulos que posee cada variable. Fácilmente detectamos varios valores nulos que posteriormente deberemos corregir.

### Resumen estadístico

In [5]:
df_titanic.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


In [6]:
df_titanic.describe(include=['object','category'])

,Name,Sex,Ticket,Cabin,Embarked
count,891,891,891,204,889
unique,891,2,681,147,3
top,"Dooley, Mr. Patrick",male,347082,G6,S
freq,1,577,7,4,644


De aquí podemos obtener distintas conclusiones:
* La variable `age` va de 0.42 hasta 80. La media es 29.7 y la mediana 28, lo que indica una leve asimetría hacia la derecha. También podemos decir que el caso del 0.42 refiere a un bebé de menos de un año.
* La variable `SibSp` tiene un cuartil 3 igual a 0, lo que indica que la gran mayoría de los pasajeros viajaron sin hermanos ni cónyuges. Sin embargo, existen pocos casos que sí han viajado con hermanos/cónyuges, incluso llegando a una cantidad máxima de 8. Algo similar ocurre con `Parch`.
* Las variables `sex` y `Embarked` son las varibles cualitativas que contienen una cantidad coherente de valores únicos. Las otras variables deberán ser examinadas para evaluar la información que nos puede aportar.

Pensá que otra conclusión se puede extraer de este resultado.

## Limpieza de datos

Ahora vamos a realizar la limpieza de los datos para limpiar todos los errores e imperfecciones que contenga el dataset.

### 0- Preparativos

Primero vamos a hacer una copia del dataset original para dejarlo como respaldo. Es una buena practica cuando se está modificando los datos. En caso de cometer algún error o inconveniente, se puede volver al dataset original y recuperar estos datos.

In [7]:
df_titanic_original = df_titanic.copy()

También, crearemos un documento que llamaremos "Bitácora" para dejar constancia del problema detectado, las variables afectadas, la decisión tomada y la justificación de dicha decisión.

In [8]:
registros_bitacora = []

def registrar(problema, variable, decision, justificacion):
    registros_bitacora.append({
        "problema_detectado": problema,
        "variable_afectada": variable,
        "decision_tomada": decision,
        "justificacion": justificacion
    })

### 1- Detección de valores duplicados

Siempre hay que revisar que no existan registros duplicados dentro del dataset. En caso de que existan, entonces habrá que eliminarlos para que evitar tener datos sesgados. Pero cuidado, registros duplicados no siempre significa mediciones registradas varias veces; puede ocurrir que -tomando este contexto- dos personas diferentes hayan conseguido la misma clase al misma precio y hyan sido acompañado por la misma cantidad de hermanos, padres e hijos. En estos casos, hay que mantener esta duplicación de filas ya que no se trata de registros registrados dos veces sino de dos individuos con las mismas características.

En nuestro caso, como tenemos la columna PassengerId, consideraremos registros duplicados facilmente si todas las columnas son iguales.

In [9]:
df_titanic.duplicated().sum()

np.int64(0)

observamos que no existen valores duplicados. Por lo tanto, no necesitamos realizar ninguna acción en este paso y contiuamos con el resto de la limpieza de datos.

### 2- Selección de variables

Como primer paso, volveremos a ver el dataframe y evaluaremos si existen columnas que no aporten información a la hora de realizar el análisis de los datos.

In [10]:
df_titanic.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


Realicemos el análisis de la variable `PassengerId`. Parece representar un identificador de los pasajeros, pero pasemos a corroborar esta suposición.

In [11]:
# Observo la cantidad de valores únicos que contiene la variable PassengerId
len(df_titanic['PassengerId'].unique())

891

Esta variable contiene 891 valores únicos. Pero el dataset recibido tiene 891 registros, por lo tanto esta variable es un identificador con valores únicos que no tienen ninguna relación con las demás variables. En conclusión, esta variable debe ser eliminada.

Lo mismo ocurre con la columna `Name`, ya que el nombre no suele ser un valor importante al momento de analizar

Por último, analizando la variable `Ticket` vemos que cae exactamente en el mismo inconveniente. Es un identificador que no aporta información.

In [12]:
# Tomamos una muestra aleatoria de los tickets
df_titanic['Ticket'].sample(20)

,Ticket
73,2680
812,28206
462,111320
295,PC 17612
274,370375
616,347080
248,11751
22,330923
134,C.A. 29178
149,244310


En definitiva, se deberán eliminar las variables `PassengerId`, `Name` y `Ticket`

In [13]:
# Elimino las columnas no deseadas. El parámetro inplace=True permite sobrescribir los cambios sobre el mismo dataframe
df_titanic.drop(columns=['PassengerId','Name','Ticket'], inplace=True)

Registramos el cambio en la bitácora

In [14]:
registrar("Selección de variables para el análisis",
          "PassengerId-Name-Ticket",
          "Eliminación de las columnas",
          "Estas columnas contienen nombres y códigos de identificación, por lo que no aportan ningún tipo de información")

### 3- Tratamiento de valores faltantes o nulos

Ahora veamos cuantos valores nulos existen en este dataset

In [15]:
df_titanic.isna().sum()

,0
Survived,0
Pclass,0
Sex,0
Age,177
SibSp,0
Parch,0
Fare,0
Cabin,687
Embarked,2


Observamos que 3 variables tienen valores nulos: `Age`, `Cabin` y `Embarked`

En el caso de `Cabin` vemos que tiene una cantidad muy elevada de valores nulos. Podemos hacer un pequeño cálculo para ver la proporción de valores faltantes.

In [16]:
proporcion_nulos = df_titanic['Cabin'].isna().sum()/df_titanic.shape[0]
print(f'La proporción de valores nulos es de {round(proporcion_nulos*100,2)}%')

La proporción de valores nulos es de 77.1%


La columna `Cabin` tiene ¡77% de valores faltantes! Es decir, casi 4 de cada 5 filas no contiene valores. Como la tasa de pérdida de información es elevada y no se observa una relación coherente sobre esta pérdida de valores, entonces eliminamos la columna completa del análisis.

In [17]:
df_titanic.drop(columns=['Cabin'], inplace=True)

# Validamos que se haya realizado correctamente la modificación
df_titanic.head()

,Survived,Pclass,Sex,Age,SibSp,Parch,Fare,Embarked
0,0,3,male,22.0,1,0,7.2500,S
1,1,1,female,38.0,1,0,71.2833,C
2,1,3,female,26.0,0,0,7.9250,S
3,1,1,female,35.0,1,0,53.1000,S
4,0,3,male,35.0,0,0,8.0500,S


In [18]:
registrar("Tratamiento de valores faltantes",
          "Cabin",
          "Eliminación de la columna",
          "La columna contiene 77% de valores faltantes, por lo que no aporta datos suficientes para obtener información")

Ahora veamos la variable `Embarked`. Ésta tiene 2 valores nulos, una cantidad muy baja. Podríamos asignarle algún valor específico, pero este valor lo asignamos nosotros y puede no ser el real. Esto ensucia el dataset cuando en realidad queremos limpiarlo. Por lo tanto, la mejor opción que podemos tomar es eliminar estas dos filas y no considerarlas en el análisis.

**Importante:** la eliminación de las filas que contienen valores nulos en al menos una columna es útil cuando esta cantidad no es considerable (en la mayoría de los casos, podemos hablar de aproximadamente 10% del total de registros del dataset).

In [19]:
df_titanic.dropna(subset=['Embarked'], inplace=True)

# Validamos la modificación
df_titanic.isna().sum()

,0
Survived,0
Pclass,0
Sex,0
Age,177
SibSp,0
Parch,0
Fare,0
Embarked,0


In [20]:
registrar("Tratamiento de valores faltantes",
          "Embarked",
          "Eliminación de las filas",
          "Existen solamente 2 filas con nulos en la variable. Es una cantidad insignificante y la eliminación de estos registros no modificará el desarrollo del análisis")

Ahora nos queda solamente la edad.

In [21]:
proporcion_nulos = df_titanic['Age'].isna().sum()/df_titanic.shape[0]
print(f'La proporción de valores nulos es de {round(proporcion_nulos*100,2)}%')

La proporción de valores nulos es de 19.91%


En este caso, la proporción no es muy elevada (como en `Cabin`) pero no es tan baja (como en `Embarked`). Si queremos eliminar las filas con valores nulos estaríamos perdiende la quinta parte de todos los datos, una cantidad muy elevada de información. Por lo tanto, recurriremos a otras técnicas.

Una posibilidad es imputar los valores nulos con algún valor. Como no conocemos qué valores pudieron tener realmente, entonces le asignamos algún valor que no afecte al comportamiento en general. Las opciones son la media, si la variable tiene una distribución simétrica de datos; la mediana, si la variable tiene una distribución muy asímetrica; y la moda, si la variable es de tipo cualitativa.

La otra posibilidad es segmentar esta variable en distintas clases. De esta forma, quedarían categorías como "bebé", "adolescente", "adulto", "anciano" y "sin dato". Estaremos perdiendo precisión, pero mantenemos todos los registros en el análisis de datos.

Para este caso, decidiremos imputar los nulos de la variable `Age` por la media.

In [22]:
df_titanic['Age'].fillna(df_titanic['Age'].mean(), inplace=True)

# Validamos la modificación
df_titanic.isna().sum()

/tmp/ipykernel_1797/4243601010.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df_titanic['Age'].fillna(df_titanic['Age'].mean(), inplace=True)


,0
Survived,0
Pclass,0
Sex,0
Age,0
SibSp,0
Parch,0
Fare,0
Embarked,0


In [23]:
registrar("Tratamiento de valores faltantes",
          "Age",
          "imputación por la media",
          "Casi el 20% de los valores son faltantes. Es poco para eliminar la columna y es mucho para eliminar las filas. Así que se decide imputar por la media, ya que presenta una distribución levemente simétrica")

### 4- Tratamiento de valores erroneos

Ahora hay que valor que los datos que tenemos ahora son válidos y correctos. Para eso, deberemos usar un poco de nuestra razón y sentido común para encontrar posibles errores. Rápidamente podemos notar que todas las variables, excepto `Fare`, contienen valores posibles.

In [24]:
for i in ['Survived','Pclass','Sex','SibSp','Parch','Embarked']:
    print(f'Los valores únicos de la variable {i} son: {df_titanic[i].unique()}')

Los valores únicos de la variable Survived son: [0 1]
Los valores únicos de la variable Pclass son: [3 1 2]
Los valores únicos de la variable Sex son: ['male' 'female']
Los valores únicos de la variable SibSp son: [1 0 3 4 2 5 8]
Los valores únicos de la variable Parch son: [0 1 2 5 3 4 6]
Los valores únicos de la variable Embarked son: ['S' 'C' 'Q']


In [25]:
df_titanic['Age'].describe()

,Age
count,889.000000
mean,29.642093
std,12.968346
min,0.420000
25%,22.000000
50%,29.642093
75%,35.000000
max,80.000000


Observemos el caso de la variable `Fare`

In [26]:
df_titanic['Fare'].describe()

,Fare
count,889.000000
mean,32.096681
std,49.697504
min,0.000000
25%,7.895800
50%,14.454200
75%,31.000000
max,512.329200


Podemos notar dos valores sospechosos:

* Fare = 0. Esto significaría que el pasaje tuvo un costo de 0, lo cual no parece algo coherente. Incluso, este valor se relaciona con diferente valores de `Pclass`, lo cual tampoco tiene sentido. Por ende, consideraremos el valor 0 como un error.
* Fare = 512. Este valor, aunque es posible, es muy diferente de los demás. Encima, el segundo boleto más caro costó 263. Lo que puede aumentar más las sospechas. Sin embargo, como estamos hablando del titanic que tenía camarotes exclusivos de primera clase, podemos pensar que este valor pertenece a este tipo de boletos. Por lo tanto, lo consideraremos como un valor atípico y no como un error.



In [27]:
df_titanic[(df_titanic['Fare']==0)|(df_titanic['Fare']>500)]

,Survived,Pclass,Sex,Age,SibSp,Parch,Fare,Embarked
179,0,3,male,36.000000,0,0,0.0000,S
258,1,1,female,35.000000,0,0,512.3292,C
263,0,1,male,40.000000,0,0,0.0000,S
271,1,3,male,25.000000,0,0,0.0000,S
277,0,2,male,29.642093,0,0,0.0000,S
302,0,3,male,19.000000,0,0,0.0000,S
413,0,2,male,29.642093,0,0,0.0000,S
466,0,2,male,29.642093,0,0,0.0000,S
481,0,2,male,29.642093,0,0,0.0000,S
597,0,3,male,49.000000,0,0,0.0000,S


Para sanar el error interpretado del Fare=0, eliminaremos las filas (ya que son pocas las filas afectadas)

In [28]:
df_titanic = df_titanic[~(df_titanic['Fare']==0)]

# Validamos la modificación imprimiendo el resultado de la expresión para ver los registros restantes
df_titanic['Fare'].describe()

,Fare
count,874.000000
mean,32.647539
std,49.942710
min,4.012500
25%,7.925000
50%,14.500000
75%,31.275000
max,512.329200


In [29]:
registrar("Tratamiento de valores erroneos",
         "Fare",
          "Eliminación de filas con Fare == 0",
          "Se considera un Fare igual a 0 como un error. Se eliminan las filas por ser una proporción pequeña de registros")

### 5- Detección de valores atípicos

Por último, veamos si podemos considerar algunos valores como outliers. Recordamos que un outlier o valor atípico es aquel que se encuentra alejado del comportamiento del resto de los datos y, por consiguiente, se lo considera fuera de la norma o un caso extraño.

Pero guarda, estos casos no deben considerarse errores, ya que un valor fuera de lo común puede ocurrir y ser completamente válido. Por lo tanto, estos casos no deben ser tratados; simplemente daremos a conocer estas existencias para más adelante.

Para esto, utilizaremos una función que nos permita calcular un outlier mediante las dos siguientes fórmulas:


In [30]:
def detectar_outliers(serie, method='iqr'):
    if method=='iqr':
      q1 = serie.quantile(0.25)
      q3 = serie.quantile(0.75)
      iqr = q3 - q1
      limite_inferior = q1 - 1.5 * iqr
      limite_superior = q3 + 1.5 * iqr
    elif method=='3sig':
      media = serie.mean()
      desviacion_estandar = serie.std()
      limite_inferior = media - 3 * desviacion_estandar
      limite_superior = media + 3 * desviacion_estandar
    else:
      raise ValueError("El método debe ser 'iqr' o '3sig'.")

    outliers = serie[(serie < limite_inferior) | (serie > limite_superior)]
    return outliers

In [31]:
for i in ['Age','SibSp','Parch','Fare']:
    print(f'Cantidad de valores atípicos de la variable {i}: {detectar_outliers(df_titanic[i]).count()}')


Cantidad de valores atípicos de la variable Age: 65
Cantidad de valores atípicos de la variable SibSp: 46
Cantidad de valores atípicos de la variable Parch: 213
Cantidad de valores atípicos de la variable Fare: 114


In [32]:
# Visualización de los valores atípicos de una variable
detectar_outliers(df_titanic['Age']).sort_values()

,Age
803,0.42
755,0.67
644,0.75
469,0.75
78,0.83
...,...
116,70.50
96,71.00
493,71.00
851,74.00


## Resumen de limpieza de datos

In [33]:
pd.DataFrame(registros_bitacora)

,problema_detectado,variable_afectada,decision_tomada,justificacion
0,Selección de variables para el análisis,PassengerId-Name-Ticket,Eliminación de las columnas,Estas columnas contienen nombres y códigos de ...
1,Tratamiento de valores faltantes,Cabin,Eliminación de la columna,"La columna contiene 77% de valores faltantes, ..."
2,Tratamiento de valores faltantes,Embarked,Eliminación de las filas,Existen solamente 2 filas con nulos en la vari...
3,Tratamiento de valores faltantes,Age,imputación por la media,Casi el 20% de los valores son faltantes. Es p...
4,Tratamiento de valores erroneos,Fare,Eliminación de filas con Fare == 0,Se considera un Fare igual a 0 como un error. ...


## Conclusión

Finalizamos así la limpieza del dataset. De esta manera, conseguimos una tabla sin problemas, lista para pasar por las próximas fases del análisis de datos.

Es importante realizar siempre la limpieza de datos al momento de recibir un conjunto de datos a estudiar. Los pasos presentados son los mínimos e indispensables para el análisis de datos. Existen muchas técnicas más para la limpieza de los datos, pero con esto será suficiente para avanzar con el Trabajo Práctico Integrador y con la materia.

# Práctica autónoma

Ahora vas a trabajar con un segundo conjunto de datos para que realices la limpieza correspondiente y construyas una bitácora propia.


## Dataset: Google Play Store Apps

Trabajaremos con el dataset 'Google Play Store Apps'. Este conjunto de datos contiene información sobre aplicaciones disponibles en Google Play Store, incluyendo categorías, calificaciones, reseñas, tamaños, número de instalaciones y más. Es útil para analizar tendencias del mercado de aplicaciones, la popularidad de ciertas categorías o el impacto de las características en las calificaciones de los usuarios.

Tu objetivo será limpiar todos los problemas que encuentres y dejarlo listo para un análisis profundo. Tené en cuenta los pasos hechos para limpiar el dataset titanic.

Aquí se describen en detalle las variables del dataset:

| Variable         | Descripción                                                                 |
| :--------------- | :-------------------------------------------------------------------------- |
| `App`            | Nombre de la aplicación.                                                    |
| `Category`       | Categoría a la que pertenece la aplicación (e.g., 'ART_AND_DESIGN', 'GAME'). |
| `Rating`         | Calificación promedio de la aplicación por los usuarios (1.0-5.0).           |
| `Reviews`        | Número de reseñas de usuarios para la aplicación.                            |
| `Size`           | Tamaño de la aplicación.                                                    |
| `Installs`       | Número de veces que la aplicación ha sido instalada.                         |
| `Type`           | Tipo de aplicación (e.g., 'Free', 'Paid').                                  |
| `Price`          | Precio de la aplicación (en USD, si es de pago).                             |
| `Content Rating` | Clasificación de contenido de la aplicación (e.g., 'Everyone', 'Teen').     |
| `Genres`         | Géneros a los que pertenece la aplicación.                                  |
| `Last Updated`   | Fecha de la última actualización de la aplicación.                           |
| `Current Ver`    | Versión actual de la aplicación.                                            |
| `Android Ver`    | Versión mínima de Android requerida para la aplicación.                     |


Con la siguiente celda podrás obtener el dataset completo. Si llega a generar errores o problemas, podés descargar el dataset que está subido en el aula virtual, subirlo a la carpeta "Archivos" de colab y ejecutar el siguiente código:

```
df_playstore = pd.load_csv('\content\google-play-store-apps.csv')
```



In [34]:
import kagglehub
import os

# Download latest version
path = kagglehub.dataset_download("lava18/google-play-store-apps")
dataset_path = os.path.join(path, 'googleplaystore.csv')
df_playstore = pd.read_csv(dataset_path)

Using Colab cache for faster access to the 'google-play-store-apps' dataset.


In [35]:
df_playstore.head()

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
0,Photo Editor & Candy Camera & Grid & ScrapBook,ART_AND_DESIGN,4.1,159,19M,"10,000+",Free,0,Everyone,Art & Design,"January 7, 2018",1.0.0,4.0.3 and up
1,Coloring book moana,ART_AND_DESIGN,3.9,967,14M,"500,000+",Free,0,Everyone,Art & Design;Pretend Play,"January 15, 2018",2.0.0,4.0.3 and up
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",ART_AND_DESIGN,4.7,87510,8.7M,"5,000,000+",Free,0,Everyone,Art & Design,"August 1, 2018",1.2.4,4.0.3 and up
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644,25M,"50,000,000+",Free,0,Teen,Art & Design,"June 8, 2018",Varies with device,4.2 and up
4,Pixel Draw - Number Art Coloring Book,ART_AND_DESIGN,4.3,967,2.8M,"100,000+",Free,0,Everyone,Art & Design;Creativity,"June 20, 2018",1.1,4.4 and up



## Consignas

1. Creá una copia de trabajo y mantené intacto el dataset original.
2. Revisá dimensiones de dataset, tipos de varaiables y estadísticos básicos.
3. Realizá todos los pasos para la limpieza de datos. Recordá siempre mantener un registro de todas las modificaciones hechas en el dataset.
4. Finalizá con una breve conclusión de cómo viste el dataset, qué cosas encontraste, qué modificaciones tuviste que realizar y por qué.

Al finaìzar, revisá los siguientes puntos para corroborar que todo estuvo ok.

- [ ] Trabajé sobre una copia.
- [ ] Diferencié faltantes, inconsistencias, duplicados y posibles valores atípicos.
- [ ] No eliminé registros únicamente porque una regla los señaló.
- [ ] Justifiqué cada cambio.
- [ ] Completé la bitácora.
- [ ] El notebook puede ejecutarse desde el comienzo.